modules/RTC/TrackStreamingStatus.js

```diff
@@ -0,0 +1,758 @@
+ *
+ * @type {number}
+ */
+const DEFAULT_RTC_MUTE_TIMEOUT = 10000;
+
+/**
+ * The time to wait a track to be restored. Track which was out of forwarded
+ * sources should be inactive and when entering forwarded sources it becomes
+ * restoring and when data is received from bridge it will become active,
+ * but if no data is received for some time we set status of that track
+ * connection to interrupted.
+ * @type {number}
+ */
+const DEFAULT_RESTORING_TIMEOUT = 10000;
+
+/**
+ * Participant connection statuses.
+ *
+ * @type {{
+ *      ACTIVE: string,
+ *      INACTIVE: string,
+ *      INTERRUPTED: string,
+ *      RESTORING: string
+ * }}
+ */
+export const TrackStreamingStatus = {
+    /**
+     * Status indicating that connection is currently active.
+     */
+    ACTIVE: 'active',
+
+    /**
+     * Status indicating that connection is currently inactive.
+     * Inactive means the connection was stopped on purpose from the bridge,
+     * like exiting forwarded sources or adaptivity decided to drop video because of not
+     * enough bandwidth.
+     */
+    INACTIVE: 'inactive',
+
+    /**
+     * Status indicating that connection is currently interrupted.
+     */
+    INTERRUPTED: 'interrupted',
+
+    /**
+     * Status indicating that connection is currently restoring.
+     */
+    RESTORING: 'restoring'
+};
+
+/**
+ * Class is responsible for emitting
+ * JitsiTrackEvents.TRACK_STREAMING_STATUS events.
+ */
+export default class TrackStreamingStatusHandler {
+    /* eslint-disable max-params*/
+    /**
+     * Calculates the new {@link TrackStreamingStatus} based on
+     * the values given for some specific remote track. It is assumed that
+     * the conference is currently in the JVB mode (in contrary to the P2P mode)
+     * @param {boolean} isInForwardedSources indicates whether the track is in
+     * the forwarded sources set. When set to false it means that JVB is not
+     * sending any video for the track.
+     * @param {boolean} isRestoringTimedout if true it means that the track has
+     * been outside of forwarded sources too long to be considered
+     * {@link TrackStreamingStatus.RESTORING}.
+     * @param {boolean} isVideoMuted true if the track is video muted and we
+     * should not expect to receive any video.
+     * @param {boolean} isVideoTrackFrozen if the current browser support video
+     * frozen detection then it will be set to true when the video track is
+     * frozen. If the current browser does not support frozen detection the it's
+     * always false.
+     * @return {TrackStreamingStatus} the new connection status for
+     * the track for whom the values above were provided.
+     * @private
+     */
+    static _getNewStateForJvbMode(
+            isInForwardedSources,
+            isRestoringTimedout,
+            isVideoMuted,
+            isVideoTrackFrozen) {
+
+        // We are currently not checking the endpoint connection status received from the JVB.
```

Comment:
The logic to store (`connStatusFromJvb`) and check the endpoint connection status from JVB have been removed.